Browser_IDE/notifications.js

```diff
@@ -0,0 +1,72 @@
+"use strict";
+
+const NotificationIcons = Object.freeze({
+  CONSTRUCTION:   { class: "bi-box-seam-fill", color: "#FFF" },
+  ERROR:  { class: "bi-exclamation-octagon", color: "#F00" },
+  WARNING: { class: "bi-exclamation-triangle", color: "#FA0" },
+  SUCCESS: { class: "bi-patch-check", color: "#0F0" },
+  INFO: { class: "bi-exclamation-triangle", color: "#FFF" },
+  NONE: { class: "", color: "#FFF" }
+});
+
+// setup area for notifications to appear in
+let notificationsArea = elem("div", { class: "sk-notification-area" });
+document.body.appendChild(notificationsArea);
+
+
+function displayEditorNotification(message, icon=NotificationIcons.NONE, timeout=5, callback=null){
+    // construct notification
+    let notificationIcon = elem("span", { class: icon.class, style: {color: icon.color} });
+
+    let notificationText = elem("div", { class: "sk-notification-body" }, [elemFromText(message)]);
+
+    let notificationCloseButton = elem("button", { class: "bi bi-x-lg" });
+    let notificationCloseButtonWrapper = elem("div", {}, [notificationCloseButton]);
+
+    let notification = elem("div", {
+        class: "sk-contents sk-notification",
+        style: {
+            cursor: callback != null ? "pointer" : ""
+        }
+    }, [
+        notificationIcon,
+        notificationText,
+        notificationCloseButtonWrapper
+    ]);
+
+    // setup functions for interaction
+    let deleteNotification = function (){
+        notification.style.pointerEvents = "none";
+        notification.style.opacity = 0;
+
+        clearTimeout(timeoutFunc);
+
+        setTimeout(function(){
+            notificationsArea.removeChild(notification);
+        }, 400 /*synchronized to end just before css transition*/);
```

Comment:
Good idea - have switched to using that. I think there's something similar in the treeview when handling the folder expansion animation - should fix at some point.
Removing the element before it becomes fully transparent was to avoid frames where the notification is (practically) invisible yet still taking space - I found it felt more responsive. Have added a comment explaining why a bit better 😃